lib/LanguageServerRename/Adapter/Worse/VariableRenamer.php

```diff
@@ -2,194 +2,27 @@
 
 namespace Phpactor\Extension\LanguageServerRename\Adapter\Worse;
 
-use Generator;
 use Microsoft\PhpParser\Node;
-use Microsoft\PhpParser\Node\ArrayElement;
 use Microsoft\PhpParser\Node\Expression\Variable;
 use Microsoft\PhpParser\Node\Parameter;
 use Microsoft\PhpParser\Node\PropertyDeclaration;
-use Microsoft\PhpParser\Node\QualifiedName;
-use Microsoft\PhpParser\Node\Statement\ForeachStatement;
-use Microsoft\PhpParser\Node\StringLiteral;
-use Microsoft\PhpParser\Parser;
-use Microsoft\PhpParser\Token;
-use Phpactor\Extension\LanguageServerRename\Model\RenameResult;
-use Phpactor\Extension\LanguageServerRename\Model\Renamer;
-use Phpactor\TextDocument\ByteOffset;
 use Phpactor\TextDocument\ByteOffsetRange;
-use Phpactor\TextDocument\Location;
-use Phpactor\TextDocument\TextDocument;
-use Phpactor\TextDocument\TextDocumentLocator;
-use Phpactor\TextDocument\TextDocumentUri;
-use Phpactor\TextDocument\TextEdit;
-use Phpactor\TextDocument\TextEdits;
 
-class VariableRenamer implements Renamer
+class VariableRenamer extends AbstractReferenceRenamer
```

Comment:
@BladeMF I fixed the variable definition locator, now both the member and variable renamers differ only with the detection of the range

Reply:
Good, that was _one_ solution. I shied away as I am not certain that all renamers will have this logic, but calling it `ReferenceRenamer` is probably a good strategy!